File: WebApp/WebApp/Models/PriceListItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApp.Models
{
    public class PriceListItem
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public decimal BasePrice { get; set; }
        
        [Required]
        [ForeignKey("TicketType")]
        public int TicketTypeId { get; set; }
        public virtual TicketType TicketType { get; set; }

        [Required]
        [ForeignKey("PriceList")]
        public int PriceListId { get; set; }
        public virtual PriceList PriceList { get; set; }

        [ForeignKey("Discount")]
        public int DiscountId { get; set; }
        public virtual UserTypeDiscount Discount { get; set; }


    }
}

File: NodeJsApp/src/models/priceListItem.js
import { Schema, model } from 'mongoose';

const priceListItemSchema = new Schema({
    basePrice: {
        type: Number,
        required: true
    },
    ticketType: {
        type: Schema.Types.ObjectId,
        ref: 'TicketType',
        required: true
    },
    priceList: {
        type: Schema.Types.ObjectId,
        ref: 'PriceList',
        required: true
    },
    discount: {
        type: Number,
        default: 0
    },
    userType: {
        type: Schema.Types.ObjectId,
        ref: 'UserType',
        required: true
    }
});

export default model('PriceListItem', priceListItemSchema);